python/cudf/cudf/_lib/pylibcudf/null_mask.pyx

```diff
@@ -0,0 +1,157 @@
+# Copyright (c) 2024, NVIDIA CORPORATION.
+
+from libcpp.memory cimport make_unique, unique_ptr
+from libcpp.pair cimport pair
+from libcpp.utility cimport move
+
+from rmm._lib.device_buffer cimport DeviceBuffer, device_buffer
+
+from cudf._lib.pylibcudf.libcudf cimport null_mask as cpp_null_mask
+from cudf._lib.pylibcudf.libcudf.types cimport mask_state, size_type
+
+from cudf._lib.pylibcudf.libcudf.types import \
+    mask_state as MaskState  # no-cython-lint
+
+from .column cimport Column
+from .table cimport Table
+
+
+cpdef DeviceBuffer copy_bitmask(Column col):
+    """Copies ``col``'s bitmask into a ``DeviceBuffer``.
+
+    For details, see :cpp:func:`copy_bitmask`.
+
+    Parameters
+    ----------
+    col : Column
+        Column whose bitmask needs to be copied
+
+    Returns
+    -------
+    rmm.DeviceBuffer
+        A ``DeviceBuffer`` containing ``col``'s bitmask, or an empty ``DeviceBuffer``
+        if ``col`` is not nullable
+    """
+    cdef device_buffer db
+    cdef unique_ptr[device_buffer] up_db
+
+    with nogil:
+        db = move(cpp_null_mask.copy_bitmask(col.view()))
+        up_db = move(make_unique[device_buffer](move(db)))
```

Comment:
You might be able to reduce the number of moves here by inlining, but I'm not sure. You _may_ be able to remove the innermost move in my suggestion below, but I doubt it.
```suggestion
        up_db = move(make_unique[device_buffer](move(cpp_null_mask.copy_bitmask(col.view()))))
```